src/backend/mod.rs

```diff
@@ -3,6 +3,7 @@
 pub mod kvm;
 pub mod sev;
 pub mod sgx;
+mod vm_base;
```

Comment:
I'm not sure I'm a fan of this. Can we just export these types from the `kvm` module?